README.md

```diff
@@ -1,64 +1,1409 @@
+- [backup](./docs/resources/backup.md) (not support update)
+- [restore](./docs/resources/restore.md) (not support update and delete)
+
+DataSource
+- [project](./docs/data-sources/project.md)
+- [cluster spec](./docs/data-sources/cluster_spec.md)
+- [restore](./docs/data-sources/restore.md)
+- [backcup](./docs/data-sources/backup.md)
+
+
+## Using the provider
+
+Documentation about the provider use and the corresponding specific configuration options can be found on the [official's website](https://www.terraform.io/language/providers).
+
+Here We just give an example to show how to use the TiDBCloud provider. 
+
+In this example, you will create and manage a dedicated cluster, create a backup for it and restore from the backup.
+
+### Set up
+
+TiDBCloud provider has released to terraform registry. All you need to do is install terraform (>=1.0).
+
+For Mac user, you can install it with Homebrew.
 
+First, install the HashiCorp tap, a repository of all our Homebrew packages.
 ```shell
-go install
+brew install hashicorp/tap/terraform
+```
+Now, install Terraform with hashicorp/tap/terraform.
+```shell
+brew install hashicorp/tap/terraform
+```
+
+See [official doc](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/aws-get-started) for other installation methods.
+
+### Create an API key
+
+The TiDB Cloud API uses HTTP Digest Authentication. It protects your private key from being sent over the network. 
+
+However, terraform-provider-tidbcloud does not support managing API key now. So you need to create the API key in the [console](https://tidbcloud.com/console/clusters).
+
+1. Click the account name in the upper-right corner of the TiDB Cloud console
+2. Click Organization Settings. The organization settings page is displayed
+3. Click the API Keys tab and then click Create API Key
+4. Enter a description for your API key. The role of the API key is always Owner currently
+5. Click Next. Copy and save the public key and the private key
+6. Make sure that you have copied and saved the private key in a secure location. The private key only displays upon the creation. After leaving this page, you will not be able to get the full private key again
+7. Click Done
+
+### Get TiDBCloud provider
+
+Create a main.tf file to get the TiDBCloud provider with API key:
+
+```
+terraform {
+  required_providers {
+    tidbcloud = {
+      source = "tidbcloud/tidbcloud"
+      version = "~> 1.0.0"
+    }
+  }
+  required_version = ">= 1.0.0"
+}
 ```
+- The `source` attribute defines the provider which will be downloaded from [Terraform Registry](https://registry.terraform.io/?_gl=1*r7mwys*_ga*NTgyMzMzOTU1LjE2NTk5NDA0MTA.*_ga_P7S46ZYEKW*MTY2MTQ4MTg5Mi4yNy4xLjE2NjE0ODIxODcuMC4wLjA.) by default
+- The `version` attribute is optional which defines the version of the provider, it will use the latest version in default
+- The `required_version` is optional which defines the version of the terraform, it will use the latest version in default
```

Comment:
```suggestion
- The `version` attribute is optional which defines the version of the provider, it will use the latest version by default
- The `required_version` is optional which defines the version of the terraform, it will use the latest version by default
```